primitives/src/common.rs

```diff
@@ -51,20 +43,22 @@ impl FromStr for PalletString {
 	}
 }
 
-pub trait AsByteOrNoop {
-	fn as_bytes_or_noop(&self) -> &[u8];
+pub trait FromCropping: Sized {
+	fn from_cropping(data: Vec<u8>) -> Self;
 }
 
-impl AsByteOrNoop for UnboundedPalletString {
-	#[cfg(feature = "std")]
-	fn as_bytes_or_noop(&self) -> &[u8] {
-		self.as_bytes()
+impl FromCropping for PalletString {
+	fn from_cropping(data: Vec<u8>) -> Self {
+		if data.len() <= Self::bound() {
+			Self::try_from(data).expect("can't be too long")
+		} else {
+			Self::try_from(data[..Self::bound()].to_vec()).expect("can't be too long now")
+		}
```

Comment:
For bounded vec they use the term `truncating_from`, I suggest we use the same: https://paritytech.github.io/substrate/master/frame_support/storage/bounded_vec/struct.BoundedVec.html#method.truncate_from

Reply:
Haha, was looking for that but didn't find it in the docs....